### PR TITLE
[socket] handle `connectError`

### PR DESCRIPTION
**Pull request by alexlamsl**
assorted clean-ups & fixes

### CODE REVIEW

File: src/bun.js/api/bun/socket.zig [hunk 1]
```diff
@@ -224,7 +224,9 @@ pub const SocketConfig = struct {
 
             const port_value = opts.get(globalObject, "port") orelse JSValue.zero;
             if (port_value.isEmptyOrUndefinedOrNull() or !port_value.isNumber() or port_value.toInt64() > std.math.maxInt(u16) or port_value.toInt64() < 0) {
-                exception.* = JSC.toInvalidArguments("Expected \"port\" to be a number between 0 and 65432", .{}, globalObject).asObjectRef();
+                exception.* = JSC.toInvalidArguments("Expected \"port\" to be a number between 0 and {d}", .{
```
Comment (Jarred-Sumner) [inline src/bun.js/api/bun/socket.zig]: we can do `comptimePrint` but actually just hardcode it

**Comment by Jarred-Sumner** [comment]
if you run the TCP server tests 

the one that checks the number of non-GC'd sockets still works right?

the other tricky part with this code is making sure that TCP socket gets freed when it should

**Comment by alexlamsl** [comment]
> the one that checks the number of non-GC'd sockets still works right?

Actually that one never worked on my computer before &minus; let me have a look again.

File: src/bun.js/api/bun/socket.zig (hunk 1 above)
Comment (alexlamsl) [reply src/bun.js/api/bun/socket.zig]: Done in https://github.com/oven-sh/bun/pull/1705/commits/3fdcdbe8f0bb5408e7e7e3a4b5cb9c21613a123f

**Comment by alexlamsl** [comment]
> the one that checks the number of non-GC'd sockets still works right?

Test "fixed" in https://github.com/oven-sh/bun/pull/1705/commits/3fdcdbe8f0bb5408e7e7e3a4b5cb9c21613a123f

**Comment by Jarred-Sumner** [comment]
Thanks

**Pull request merged**